scripts/gdb/lvglgdb/lvgl.py

```diff
@@ -54,139 +53,108 @@ def __next__(self):
     @property
     def len(self):
         len = 0
-        node = self.ll["head"]
+        node = self.head
         while node:
             len += 1
             node = self._next(node)
         return len
 
-    @property
-    def head(self):
-        return self.ll["head"]
-
-    @property
-    def tail(self):
-        return self.ll["tail"]
-
-    @property
-    def size(self):
-        return self.ll["n_size"]
-
 
-class LVObject(gdb.Value):
+class LVObject(Value):
     """LVGL object"""
 
-    def __init__(self, obj: gdb.Value):
+    def __init__(self, obj: Value):
         super().__init__(obj)
-        self.obj = obj
-
-    @property
-    def class_p(self):
-        return self.obj["class_p"]
 
     @property
     def class_name(self):
-        name = self.class_p["name"]
+        name = self.class_p.name
         return name.string() if name else "unknown"
 
-    @property
-    def coords(self):
-        return self.obj["coords"]
-
     @property
     def x1(self):
-        return int(self.coords["x1"])
+        return int(self.coords.x1)
 
     @property
     def y1(self):
-        return int(self.coords["y1"])
+        return int(self.coords.y1)
 
     @property
     def x2(self):
-        return int(self.coords["x2"])
+        return int(self.coords.x2)
 
     @property
     def y2(self):
-        return int(self.coords["y2"])
+        return int(self.coords.y2)
 
     @property
     def child_count(self):
-        return self.obj["spec_attr"]["child_cnt"] if self.obj["spec_attr"] else 0
+        return self.spec_attr.child_cnt if self.spec_attr else 0
 
     @property
     def childs(self):
-        if not self.obj["spec_attr"]:
+        if not self.spec_attr:
             return
 
         for i in range(self.child_count):
-            child = self.obj["spec_attr"]["children"][i]
+            child = self.spec_attr.children[i]
             yield LVObject(child)
 
     @property
     def styles(self):
         LV_STYLE_PROP_INV = 0
         LV_STYLE_PROP_ANY = 0xFF
-        count = self.obj["style_cnt"]
+        count = self.style_cnt
         if count == 0:
             return
 
-        styles = self.obj["styles"]
+        styles = super().__getitem__("styles")
```

Comment:
Why styles is different?

Reply:
because they use the same name

Reply:
But this usage is weird, not everyone will know it's special and need to use the hidden magic instead of normal `self.styles` just like others.

Reply:
But you can use self.styles to get the styles value stored in gdb.Value, because you use the same api name.

If you call self.styles here, it will cause a infinity recursive calling loop.